### PR TITLE
feat: add default option

### LINKED ISSUE #83: Force compression to client don't work
I want compression response body to client always , Whether the client have any Settings。My Client App is poor what can't set any request header, but can handle compression data.

``` js
app.get('/device', auth_identity, compression({filter: function(req, res) { return true;}}),device);
```

```
node: v4.2.2
compression: 1.6.1
express : 3.4.4
```

Why ? thanks.

### PR DESCRIPTION
**Pull request by bjohansebas**
A new option is added to determine the default behavior when `Accept-Encoding` is not sent by the client, defaulting to identity.

closes #83

### CODE REVIEW

**Comment by bjohansebas** [comment]
Hi @IamLizu, could you review this PR?

**Comment by IamLizu** [comment]
Hey @bjohansebas :wave: 

Sure, I will read the relevant issues and then review this PR :+1:

File: index.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ module.exports.filter = shouldCompress
 
 var cacheControlNoTransformRegExp = /(?:^|,)\s*?no-transform\s*?(?:,|$)/
 
+var encodingSupported = ['*', 'gzip', 'deflate', 'identity']
```
Comment (IamLizu) [inline index.js:40]: Why are we only supporting these?

Comment (bjohansebas) [reply index.js:40]: These are the options supported by the package at the moment; support for Brotli is still in progress

Comment (IamLizu) [reply index.js:40]: Okay, then I will try to get a better understanding of this and get back to you.

File: index.js [hunk 2]
```diff
@@ -51,6 +53,7 @@ function compression (options) {
   // options
   var filter = opts.filter || shouldCompress
   var threshold = bytes.parse(opts.threshold)
+  var defaultEncoding = opts.defaultEncoding || 'identity'
```
Comment (IamLizu) [inline index.js:56]: We should not use `defaultEncoding`, it is in `zlib`. Anything that creates conflict down the line in related places should be avoided.

IMO, we could use `enforceEncoding`. It creates more clarity on the fact that we are allowing an enforcement. The default behavior in content negotiation is more like _what's best for the client_.

File: index.js [hunk 3]
```diff
@@ -177,12 +180,21 @@ function compression (options) {
...
+        method = defaultEncoding === '*' ? 'gzip' : defaultEncoding
+      }
+
       // negotiation failed
       if (!method || method === 'identity') {
         nocompress('not acceptable')
         return
       }
 
+      if (opts.defaultEncoding) {
```
Comment (IamLizu) [inline index.js:194]: Yeah, this shouldn't be necessary with the `enforceEncoding` variable or something else.

**Review by IamLizu — CHANGES_REQUESTED** [review]
Hi @bjohansebas  :wave: 

I'm generally :+1:  of the concept of introducing an enforcement mechanism, as discussed in https://github.com/expressjs/compression/issues/83.

However, would you mind renaming the variable `defaultEnconfig` to something that doesn't conflict with `zlib.createGzip` or `zlib.createDeflate`? It seems this conflict may have led to the addition of the following block:

```js
if (opts.defaultEncoding) {
    opts.defaultEncoding = undefined
}
```

I suggest `enforceEncoding`, more on this below. Renaming the variable would help keep the code clean, and it’s important that we avoid such conflicts.

Thanks so much for your understanding and effort!

**Comment by bjohansebas** [comment]
@IamLizu Good catch

**Review by IamLizu — APPROVED** [review]
LGTM 👍

There's something I should address. As we know, content negotiation is handled by `negotiator` here, and it is indeed `negotiator`'s responsibility to choose the best method.

Maybe are interfering with `negotiator`'s responsibilities?

cc: @expressjs/express-tc

**Comment by bjohansebas** [comment]
@UlisesGascon ping

File: index.js [hunk 4]
```diff
@@ -177,6 +180,11 @@ function compression (options) {
       var negotiator = new Negotiator(req)
       var method = negotiator.encoding(['gzip', 'deflate', 'identity'], ['gzip'])
 
+      // if no method is found, use the default encoding
+      if (encodingSupported.indexOf(enforceEncoding) !== -1 && !req.headers['accept-encoding']) {
```
Comment (wesleytodd) [inline index.js:184]: nit: can you change the order of this? We should not check if the option is valid unless we are going to use it.

File: index.js [hunk 5]
```diff
@@ -177,6 +180,11 @@ function compression (options) {
...
+        method = enforceEncoding === '*' ? 'gzip' : enforceEncoding
```
Comment (wesleytodd) [inline index.js:185]: I dont see `*` referenced anywhere but here. Can you describe why this is necessary when you could drop this check and just have folks pass `gzip` directly for this?

**Review by wesleytodd — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (bjohansebas) [reply index.js:185]: If * is not checked and changed, the deflate encoding would be applied as it’s the last one in the check (https://github.com/expressjs/compression/blob/66c0cb759d5ea36e6253df5f2fa88e4f6b6ef8c3/index.js#L196-L198), and deflate is not the encoding we want to be applied, it should be gzip when it’s *.

File: index.js (hunk 4 above)
Comment (bjohansebas) [reply index.js:184]: sure

File: index.js (hunk 5 above)
Comment (wesleytodd) [reply index.js:185]: Sorry but I don't follow your description? To me it looks like this value for the option (coming from the calling code) could just be dropped entirely.

**Review by wesleytodd — COMMENTED** [review]
It looks to me like this was merged with a pending "change requested" review. Is that the case?

File: index.js (hunk 5 above)
Comment (bjohansebas) [reply index.js:185]: I haven't found a way to eliminate this check.

**Pull request merged**